eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/BomGenerator.java

```diff
@@ -24,27 +24,77 @@
 import java.util.regex.Matcher;
 import java.util.stream.Collectors;
 
+import static com.azure.tools.bomgenerator.Utils.ANALYZE_MODE;
 import static com.azure.tools.bomgenerator.Utils.BASE_AZURE_GROUPID;
 import static com.azure.tools.bomgenerator.Utils.AZURE_PERF_LIBRARY_IDENTIFIER;
 import static com.azure.tools.bomgenerator.Utils.AZURE_TEST_LIBRARY_IDENTIFIER;
 import static com.azure.tools.bomgenerator.Utils.EXCLUSION_LIST;
+import static com.azure.tools.bomgenerator.Utils.GENERATE_MODE;
 import static com.azure.tools.bomgenerator.Utils.POM_TYPE;
 import static com.azure.tools.bomgenerator.Utils.SDK_DEPENDENCY_PATTERN;
+import static com.azure.tools.bomgenerator.Utils.parsePomFileContent;
 
 public class BomGenerator {
     private String outputFileName;
     private String inputFileName;
     private String pomFileName;
+    private String mode;
 
     private static Logger logger = LoggerFactory.getLogger(BomGenerator.class);
 
-    BomGenerator(String inputFileName, String outputFileName, String pomFileName) {
+    BomGenerator() {
+        this.mode = GENERATE_MODE;
+    }
+
+    public String getInputFileName() {
+        return this.inputFileName;
+    }
+
+    public void setInputFileName(String inputFileName) {
         this.inputFileName = inputFileName;
+    }
+
+    public String getOutputFileName() {
+        return this.outputFileName;
+    }
+
+    public void setOutputFileName(String outputFileName) {
         this.outputFileName = outputFileName;
+    }
+
+    public String getPomFileName() {
+        return this.pomFileName;
+    }
+
+    public void setPomFileName(String pomFileName) {
         this.pomFileName = pomFileName;
     }
 
-    public void generate() {
+    public String getMode() {
+        return this.mode;
+    }
+
+    public void setMode(String mode) {
+        this.mode = mode;
+    }
+
+    public void run() {
+        switch(mode) {
+            case ANALYZE_MODE: validate();
```

Comment:
Given the overall changes to support both analyzing and generating separately, should `BomGenerator` become an abstract parent class with sub-classes used for analyzing and generating separately? Each could be `Runnable`s which validate their configuration on run to either proceed or error out on missing parameters.

Reply:
I expect there to be more tweaks needed in the next couple or so weeks and  once we have that will redesign things.